payments/schema/types.py

```diff
@@ -138,3 +149,90 @@ def resolve_product_type(self, info):
             if self.service.is_fixed_service()
             else AdditionalProductType.OPTIONAL_SERVICE
         )
+
+
+class ProductUnion(graphene.Union):
+    class Meta:
+        types = (BerthProductNode, WinterStorageProductNode)
+
+
+class LeaseUnion(graphene.Union):
+    class Meta:
+        # TODO: Add WinterStorageLeaseNode
+        types = (BerthLeaseNode,)
+
+
+class OrderNode(DjangoObjectType):
+    customer = graphene.Field("customers.schema.ProfileNode", required=True)
+    product = graphene.Field(ProductUnion)
+    lease = graphene.Field(LeaseUnion)
+    status = OrderStatusEnum(required=True)
+    comment = graphene.String()
+    price = graphene.Decimal(required=True)
+    tax_percentage = graphene.Decimal(required=True)
+    pretax_price = graphene.Decimal(required=True)
+    total_price = graphene.Decimal(required=True)
+    total_pretax_price = graphene.Decimal(required=True)
+    total_tax_percentage = graphene.Decimal(required=True)
+    due_date = graphene.Date(required=True)
+    order_lines = DjangoConnectionField("payments.schema.OrderLineNode", required=True)
+    log_entries = DjangoConnectionField(
+        "payments.schema.OrderLogEntryNode", required=True
+    )
+
+    class Meta:
+        model = Order
+        interfaces = (graphene.relay.Node,)
+        connection_class = CountConnection
+        exclude = (
+            "_product_content_type",
+            "_product_object_id",
+            "_lease_content_type",
+            "_lease_object_id",
+        )
+
+    @classmethod
+    @view_permission_required(Order)
+    def get_queryset(cls, queryset, info):
+        return super().get_queryset(queryset, info)
+
+    def resolve_order_lines(self, info, **kwargs):
+        return OrderLine.objects.filter(order=self)
+
+    def resolve_log_entries(self, info, **kwargs):
+        return OrderLogEntry.objects.filter(order=self)
+
+
+class OrderLineNode(DjangoObjectType):
+    product = graphene.Field("payments.schema.AdditionalProductNode")
```

Comment:
why is this a string import? 🤔

Reply:
I was thinking about doing _all_ the imports as string imports, but I guess it makes sense to use "normal" imports when they're on the same file.

Reply:
"normal" import is the default import, as it helps you traverse the codebase by "Go To definition" etc. and makes refactoring a tiny bit easier. string imports are reasonable when you want to avoid circular import errors

Reply:
I can't remember on which docs I read it (either python-graphene or Django) that the recommended way to handle `Field` (or `ForeignKey` 🤔 ) references was through string. I'll keep looking for it, but we can go on with the "normal" ones.

Reply:
Couldn't find the docs I was looking for :sad-kitten: so let's go on with the "normal" imports until proven otherwise.

Reply:
But :hm: (again), we were already doing it on the other types (which of course can also be changed).

https://github.com/City-of-Helsinki/berth-reservations/blob/ven-398/invoicing-api/payments/schema/types.py#L63
```python
class BerthPriceGroupNode(DjangoObjectType):
    name = graphene.String(required=True)
    products = DjangoConnectionField("payments.schema.BerthProductNode", required=True)
    default_product = graphene.Field(
        "payments.schema.BerthProductNode",
        description="Returns the first product associated to the price group "
        "that does not belong to any specific `Harbor`.",
    )
```

Reply:
I did not comment there about the import format, because I thought that it was due to circular import. If you can have a normal impor there, then we should refactor it at some point. 🙂

Reply:
Ahh, fair point. I'll refactor the ones that won't cause import errors.